storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3StorageConfig.java

```diff
@@ -209,12 +223,25 @@ public int uploadPartSize() {
         return getInt(S3_MULTIPART_UPLOAD_PART_SIZE_CONFIG);
     }
 
+    private URI s3ServiceEndpoint() {
+        final String url = getString(S3_ENDPOINT_URL_CONFIG);
+        if (url != null) {
+            try {
+                return new URI(url);
+            } catch (final URISyntaxException e) {
+                throw new ConfigException(S3_ENDPOINT_URL_CONFIG, url, "Must be a valid URI");
```

Comment:
I think this is worth adding a validator, but can be done separately.

Reply:
Yeah, lets see later if we need it, after applying one of your other comments the validator does not seem to be necessary since it will anyways lead to manual call to getter or constructor of URI.